realm/realm-library/src/main/java/io/realm/RealmQuery.java

```diff
@@ -404,6 +404,191 @@ public boolean isValid() {
         return this;
     }
 
+    // EqualToAny
+
+    /**
+     * Equal-to-any comparison.
+     *
+     * @param fieldName the field to compare.
+     * @param values the array of values to compare with.
+     * @return the query object.
+     * @throws java.lang.IllegalArgumentException if one or more arguments do not match class or field type.
+     */
+    public RealmQuery<E> in(String fieldName, String[] values) {
+        return in(fieldName, values, Case.SENSITIVE);
+    }
+
+    /**
+     * Equal-to-any comparison.
+     *
+     * @param fieldName the field to compare.
+     * @param values the array of values to compare with.
+     * @return the query object.
+     * @throws java.lang.IllegalArgumentException if one or more arguments do not match class or field type.
+     */
+    public RealmQuery<E> in(String fieldName, String[] values, Case casing) {
+        if (values == null || values.length == 0) {
+            return this;
+        }
+        equalTo(fieldName, values[0], casing);
```

Comment:
We should use `beginGroup()` and `endGroup()` since `equalTo("a") and in("b", "c", "d")` does not equal to `equalTo("a") and equalTo("b") or equalTo("c") or equalTo("d")`.